src/memefactory/server/syncer.cljs

```diff
@@ -399,27 +404,37 @@
 (defn dispatch-event
   ([ev] (dispatch-event nil ev))
   ([err {:keys [args event address block-number] :as evt}]
-   (when err
-     (log/error "Error when dispatching event" {:error err} ::dispatch-event))   
-   (when evt
-     (let [contract-key (contract-key-from-address address)
-           contract-type ({:meme-registry-db         :contract/eternal-db
-                           :param-change-registry-db :contract/eternal-db
-                           :meme-registry-fwd        :contract/meme
-                           :meme-token               :contract/meme-token
-                           :meme-auction-factory     :contract/meme-auction
-                           :meme-auction-factory-fwd :contract/meme-auction} contract-key)
-           ev (-> args
-                  (assoc :contract-address address)
-                  (assoc :event (keyword event))
-                  (update :timestamp (fn [ts]
-                                       (if ts
-                                         (bn/number ts)
-                                         (:timestamp (web3-eth/get-block @web3 block-number)))))
-                  (update :version bn/number)
-                  (assoc :block-number block-number))]
-       (log/info (str "Dispatching smart contract event "  contract-type " " (:event ev)) {:ev ev} ::dispatch-event)
-       (process-event contract-type ev)))))
+   (if err
+     (log/error "Error when dispatching event" {:error err
+                                                :filter-ids (map (fn [filt] (.-filterId filt)) @all-filters)}
+                ::dispatch-event)
+     (if-let [event (keyword event)]
+       (do
+         (when (= (last-block-number) block-number)
```

Comment:
I'm not sure this method of avoiding past events emails resend is fully reliable, what about adding metadata to the event map with where it is coming from? then if it is coming from a past event reply we can skip email send